### ISSUE TITLE
add more projects to mypy_primer

### ISSUE THREAD

**Issue by hauntsaninja** [hauntsaninja/mypy_primer#42]
https://github.com/pybind/pybind11
https://github.com/ronf/asyncssh
https://github.com/cytopia/pwncat
https://github.com/NervanaSystems/ngraph
https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/setup.cfg
https://github.com/nv-legate/cunumeric
https://github.com/geofront-auth/geofront
https://github.com/whonore/Coqtail
https://github.com/relastle/glacier
https://github.com/qutebrowser/qutebrowser
https://github.com/pola-rs/polars
https://github.com/lark-parser/lark
https://github.com/tomerfiliba/plumbum
https://github.com/PyCQA/astroid
https://github.com/PyCQA/flake8-pyi
https://github.com/scikit-build/scikit-build
https://github.com/pypa/build
https://github.com/pypa/readme_renderer
https://github.com/xaviml/controllerx
https://github.com/bimmerconnected/bimmer_connected
https://github.com/scikit-hep/particle
https://github.com/scikit-hep/vector

**Comment by hauntsaninja** [comment]
https://github.com/mongodb/mongo-python-driver reported a couple typeshed regressions

**Comment by hauntsaninja** [comment]
https://github.com/artigraph/artigraph could have caught a mypy regression

**Comment by hauntsaninja** [comment]
https://github.com/enthought/comtypes

**Comment by hauntsaninja** [comment]
https://github.com/trezor/trezor-firmware once next mypy release is out

**Comment by AlexWaygood** [comment]
Sounds like https://github.com/caronc/appraise might have caught https://github.com/python/mypy/issues/13536

**Comment by hauntsaninja** [comment]
Maybe some of the projects that are now referring to https://github.com/python/mypy/issues/731

**Comment by hauntsaninja** [comment]
https://github.com/scikit-hep/pyhf would catch https://github.com/python/mypy/issues/13813

**Comment by Avasam** [comment]
This isn't a request to add any of 'em, rather just another metric/datapoint to be aware of some of mypy's big users. Using the https://github.com/github-tooling/ghtopdep CLI tool, I listed the top 100 packages and top 100 repositories (by star count) that have mypy as a dependent on Github (because Github doesn't allow to sort "dependants"):

<details>
<summary><code>ghtopdep https://github.com/python/mypy --packages --rows 100</code></summary>
<br/>

| url                                                          | stars   | primer           |
|--------------------------------------------------------------|---------|------------------|
| https://github.com/scikit-learn/scikit-learn                 | 52K     |:heavy_check_mark:|
| https://github.com/tiangolo/fastapi                          | 50K     ||
| https://github.com/ultralytics/yolov5                        | 31K     ||
| https://github.com/certbot/certbot                           | 29K     ||
| https://github.com/mitmproxy/mitmproxy                       | 29K     |:heavy_check_mark:|
| https://github.com/Lightning-AI/lightning                    | 20K     ||
| https://github.com/sanic-org/sanic                           | 17K     ||
| https://github.com/ArchiveBox/ArchiveBox                     | 14K     ||
| https://github.com/onnx/onnx                                 | 13K     ||
| https://github.com/networkx/networkx                         | 12K     ||
| https://github.com/allenai/allennlp                          | 11K     ||
| https://github.com/Chia-Network/chia-blockchain              | 11K     ||
| https://github.com/PrefectHQ/prefect                         | 10K     |:heavy_check_mark:|
| https://github.com/matrix-org/synapse                        | 10.0K   ||
| https://github.com/spotDL/spotify-downloader                 | 9.8K    ||
| https://github.com/bloomberg/memray                          | 9.3K    ||
| https://github.com/pytest-dev/pytest                         | 9.3K    |:heavy_check_mark:|
| https://github.com/openai/DALL-E                             | 8.9K    ||
| https://github.com/MagicStack/uvloop                         | 8.9K    ||
| https://github.com/tiangolo/typer                            | 8.9K    ||
| https://github.com/aws/serverless-application-model          | 8.8K    ||
| https://github.com/apache/tvm                                | 8.6K    ||
| https://github.com/airbytehq/airbyte                         | 8.2K    ||
| https://github.com/pyro-ppl/pyro                             | 7.6K    ||
| https://github.com/tektoncd/pipeline                         | 7.4K    ||
| https://github.com/great-expectations/great_expectations     | 7.3K    ||
| https://github.com/optuna/optuna                             | 7.0K    |:heavy_check_mark:|
| https://github.com/automl/auto-sklearn                       | 6.5K    ||
| https://github.com/cupy/cupy                                 | 6.4K    ||
| https://github.com/marshmallow-code/marshmallow              | 6.3K    ||
| https://github.com/scikit-learn-contrib/imbalanced-learn     | 6.1K    ||
| https://github.com/aws/aws-sam-cli                           | 6.1K    ||
| https://github.com/sqlalchemy/sqlalchemy                     | 6.1K    ||
| https://github.com/chainer/chainer                           | 5.7K    ||
| https://github.com/pypa/pipx                                 | 5.6K    ||
| https://github.com/dagster-io/dagster                        | 5.6K    ||
| https://github.com/davidhalter/jedi                          | 5.2K    ||
| https://github.com/thu-ml/tianshou                           | 5.1K    ||
| https://github.com/sphinx-doc/sphinx                         | 4.9K    |:heavy_check_mark:|
| https://github.com/twisted/twisted                           | 4.8K    |:heavy_check_mark:|
| https://github.com/python-attrs/attrs                        | 4.5K    |:heavy_check_mark:|
| https://github.com/jpadilla/pyjwt                            | 4.4K    |:heavy_check_mark:|
| https://github.com/facebookresearch/demucs                   | 4.3K    ||
| https://github.com/vyperlang/vyper                           | 4.3K    ||
| https://github.com/mesonbuild/meson                          | 4.2K    ||
| https://github.com/pdfminer/pdfminer.six                     | 3.9K    ||
| https://github.com/ethereum/web3.py                          | 3.7K    ||
| https://github.com/online-ml/river                           | 3.7K    ||
| https://github.com/getsentry/responses                       | 3.7K    ||
| https://github.com/feast-dev/feast                           | 3.6K    ||
| https://github.com/amundsen-io/amundsen                      | 3.6K    ||
| https://github.com/quantumlib/Cirq                           | 3.6K    ||
| https://github.com/pytorch/captum                            | 3.5K    ||
| https://github.com/facebookresearch/nevergrad                | 3.4K    ||
| https://github.com/Trusted-AI/adversarial-robustness-toolbox | 3.3K    ||
| https://github.com/feeluown/FeelUOwn                         | 3.2K    ||
| https://github.com/trailofbits/manticore                     | 3.1K    |:heavy_check_mark:|
| https://github.com/Overv/outrun                              | 3.1K    ||
| https://github.com/awslabs/gluonts                           | 3.0K    ||
| https://github.com/life4/textdistance                        | 3.0K    ||
| https://github.com/ethereum/consensus-specs                  | 2.9K    ||
| https://github.com/ehForwarderBot/ehForwarderBot             | 2.8K    ||
| https://github.com/evidentlyai/evidently                     | 2.8K    ||
| https://github.com/ml-tooling/opyrator                       | 2.7K    ||
| https://github.com/QQuick/Transcrypt                         | 2.6K    ||
| https://github.com/samschott/maestral                        | 2.5K    ||
| https://github.com/mandiant/flare-floss                      | 2.4K    ||
| https://github.com/mandiant/capa                             | 2.4K    ||
| https://github.com/crossbario/autobahn-python                | 2.4K    ||
| https://github.com/facebookresearch/hiplot                   | 2.4K    ||
| https://github.com/ElementsProject/lightning                 | 2.4K    ||
| https://github.com/autorope/donkeycar                        | 2.3K    ||
| https://github.com/fastapi-users/fastapi-users               | 2.2K    ||
| https://github.com/abhinavsingh/proxy.py                     | 2.1K    ||
| https://github.com/crossbario/crossbar                       | 2.0K    ||
| https://github.com/thonny/thonny                             | 2.0K    ||
| https://github.com/DerwenAI/pytextrank                       | 1.9K    ||
| https://github.com/ethereum/py-evm                           | 1.8K    ||
| https://github.com/automl/Auto-PyTorch                       | 1.8K    ||
| https://github.com/dalibo/pg_activity                        | 1.8K    ||
| https://github.com/joowani/binarytree                        | 1.8K    ||
| https://github.com/kserve/kserve                             | 1.7K    ||
| https://github.com/graphistry/pygraphistry                   | 1.7K    ||
| https://github.com/chezou/tabula-py                          | 1.7K    ||
| https://github.com/pycontribs/jira                           | 1.6K    ||
| https://github.com/colour-science/colour                     | 1.5K    ||
| https://github.com/fepegar/torchio                           | 1.5K    ||
| https://github.com/uber/petastorm                            | 1.5K    ||
| https://github.com/Nixtla/statsforecast                      | 1.5K    ||
| https://github.com/Flexget/Flexget                           | 1.5K    ||
| https://github.com/upbit/pixivpy                             | 1.4K    ||
| https://github.com/nyu-mll/jiant                             | 1.4K    ||
| https://github.com/0xProject/0x-monorepo                     | 1.4K    ||
| https://github.com/karlicoss/promnesia                       | 1.4K    ||
| https://github.com/nubank/fklearn                            | 1.4K    ||
| https://github.com/dosisod/refurb                            | 1.4K    ||
| https://github.com/rednaga/APKiD                             | 1.4K    ||
| https://github.com/Lightning-AI/lightning-bolts              | 1.4K    ||
| https://github.com/mitmproxy/pdoc                            | 1.3K    ||
| https://github.com/microsoft/CDM                             | 1.3K    ||
</details>

<details>
<summary><code>ghtopdep https://github.com/python/mypy --repositories --rows 100</code></summary>
<br/>

| url                                                        | stars   | primer           |
|------------------------------------------------------------|---------|------------------|
| https://github.com/TheAlgorithms/Python                    | 146K    ||
| https://github.com/pallets/flask                           | 61K     ||
| https://github.com/pytorch/pytorch                         | 59K     ||
| https://github.com/home-assistant/core                     | 55K     |:heavy_check_mark:|
| https://github.com/ansible/ansible                         | 55K     ||
| https://github.com/Textualize/rich                         | 40K     |:heavy_check_mark:|
| https://github.com/commaai/openpilot                       | 36K     ||
| https://github.com/pandas-dev/pandas                       | 35K     |:heavy_check_mark:|
| https://github.com/apache/spark                            | 34K     |:heavy_check_mark:|
| https://github.com/getsentry/sentry                        | 32K     ||
| https://github.com/psf/black                               | 30K     |:heavy_check_mark:|
| https://github.com/certbot/certbot                         | 29K     ||
| https://github.com/explosion/spaCy                         | 24K     ||
| https://github.com/emscripten-core/emscripten              | 23K     ||
| https://github.com/ray-project/ray                         | 22K     ||
| https://github.com/python-poetry/poetry                    | 22K     |:heavy_check_mark:|
| https://github.com/numpy/numpy                             | 22K     ||
| https://github.com/servo/servo                             | 22K     ||
| https://github.com/streamlit/streamlit                     | 21K     |:heavy_check_mark:|
| https://github.com/tornadoweb/tornado                      | 21K     |:heavy_check_mark:|
| https://github.com/deezer/spleeter                         | 21K     ||
| https://github.com/google/jax                              | 20K     |:heavy_check_mark:|
| https://github.com/Lightning-AI/lightning                  | 20K     ||
| https://github.com/celery/celery                           | 20K     ||
| https://github.com/python-telegram-bot/python-telegram-bot | 20K     ||
| https://github.com/freqtrade/freqtrade                     | 20K     |:heavy_check_mark:|
| https://github.com/SpaceVim/SpaceVim                       | 19K     ||
| https://github.com/saleor/saleor                           | 17K     ||
| https://github.com/OpenBB-finance/OpenBBTerminal           | 17K     ||
| https://github.com/zulip/zulip                             | 17K     |:heavy_check_mark:|
| https://github.com/AtsushiSakai/PythonRobotics             | 16K     ||
| https://github.com/RasaHQ/rasa                             | 15K     ||
| https://github.com/JetBrains/intellij-community            | 14K     ||
| https://github.com/pulumi/pulumi                           | 14K     ||
| https://github.com/Textualize/textual                      | 14K     ||
| https://github.com/onnx/onnx                               | 13K     ||
| https://github.com/pallets/click                           | 13K     ||
| https://github.com/aio-libs/aiohttp                        | 13K     |:heavy_check_mark:|
| https://github.com/psf/requests-html                       | 13K     ||
| https://github.com/ankitects/anki                          | 12K     ||
| https://github.com/networkx/networkx                       | 12K     ||
| https://github.com/ManimCommunity/manim                    | 12K     ||
| https://github.com/pydantic/pydantic                       | 11K     ||
| https://github.com/allenai/allennlp                        | 11K     ||
| https://github.com/ceph/ceph                               | 11K     ||
| https://github.com/NixOS/nixpkgs                           | 10K     ||
| https://github.com/scipy/scipy                             | 10K     |:heavy_check_mark:|
| https://github.com/PrefectHQ/prefect                       | 10K     |:heavy_check_mark:|
| https://github.com/tiangolo/full-stack-fastapi-postgresql  | 10.0K   ||
| https://github.com/matrix-org/synapse                      | 10.0K   ||
| https://github.com/dbader/schedule                         | 9.9K    ||
| https://github.com/spotDL/spotify-downloader               | 9.8K    ||
| https://github.com/PostHog/posthog                         | 9.5K    ||
| https://github.com/darknessomi/musicbox                    | 9.5K    ||
| https://github.com/encode/httpx                            | 9.5K    ||
| https://github.com/bloomberg/memray                        | 9.3K    ||
| https://github.com/aws/chalice                             | 9.2K    ||
| https://github.com/pyodide/pyodide                         | 9.2K    |:heavy_check_mark:|
| https://github.com/coding-horror/basic-computer-games      | 9.1K    ||
| https://github.com/openai/DALL-E                           | 8.9K    ||
| https://github.com/pallets/jinja                           | 8.8K    |:heavy_check_mark:|
| https://github.com/aristocratos/bpytop                     | 8.8K    ||
| https://github.com/aws/serverless-application-model        | 8.8K    ||
| https://github.com/tiangolo/sqlmodel                       | 8.7K    ||
| https://github.com/pola-rs/polars                          | 8.4K    ||
| https://github.com/OpenMined/PySyft                        | 8.4K    ||
| https://github.com/airbytehq/airbyte                       | 8.2K    ||
| https://github.com/qutebrowser/qutebrowser                 | 8.2K    ||
| https://github.com/streamlink/streamlink                   | 7.9K    ||
| https://github.com/modin-project/modin                     | 7.9K    ||
| https://github.com/kedro-org/kedro                         | 7.7K    ||
| https://github.com/pytube/pytube                           | 7.6K    ||
| https://github.com/microsoft/onnxruntime                   | 7.6K    ||
| https://github.com/encode/starlette                        | 7.4K    |:heavy_check_mark:|
| https://github.com/laramies/theHarvester                   | 7.4K    ||
| https://github.com/ActivityWatch/activitywatch             | 7.3K    ||
| https://github.com/great-expectations/great_expectations   | 7.3K    ||
| https://github.com/returntocorp/semgrep                    | 7.2K    ||
| https://github.com/yugabyte/yugabyte-db                    | 7.1K    ||
| https://github.com/microsoft/playwright-python             | 6.9K    ||
| https://github.com/SirVer/ultisnips                        | 6.9K    ||
| https://github.com/microsoft/unilm                         | 6.9K    ||
| https://github.com/qemu/qemu                               | 6.8K    ||
| https://github.com/doccano/doccano                         | 6.8K    ||
| https://github.com/shidenggui/easytrader                   | 6.8K    ||
| https://github.com/zephyrproject-rtos/zephyr               | 6.7K    ||
| https://github.com/duckdb/duckdb                           | 6.7K    ||
| https://github.com/facebookresearch/hydra                  | 6.4K    ||
| https://github.com/robinhood/faust                         | 6.3K    ||
| https://github.com/HypothesisWorks/hypothesis              | 6.2K    ||
| https://github.com/pallets/werkzeug                        | 6.2K    |:heavy_check_mark:|
| https://github.com/openedx/edx-platform                    | 6.1K    ||
| https://github.com/spulec/moto                             | 6.1K    ||
| https://github.com/aws/aws-sam-cli                         | 6.1K    ||
| https://github.com/apache/doris                            | 6.1K    ||
| https://github.com/plasma-umass/scalene                    | 6.0K    ||
| https://github.com/guardicore/monkey                       | 5.9K    ||
| https://github.com/Shougo/deoplete.nvim                    | 5.8K    ||
| https://github.com/dbt-labs/dbt-core                       | 5.8K    ||
| https://github.com/encode/uvicorn                          | 5.7K    ||
</details>

**Comment by hauntsaninja** [comment]
If we want more coverage of sci code, can look at outlinks from https://github.com/python/mypy/issues/13627

**Comment by hauntsaninja** [comment]
https://github.com/mit-ll-responsible-ai/hydra-zen
https://github.com/sqlalchemy/sqlalchemy (even without plugin)

**Comment by AlexWaygood** [comment]
You can cross off flake8-pyi from the list at the top (added in https://github.com/hauntsaninja/mypy_primer/commit/893b19fb142b0953f9bf7be766e685de47e55217 :)

**Comment by AlexWaygood** [comment]
`pwndbg` might be a good project to add, since they use `types-gdb` in CI — I doubt we have any coverage of typeshed's `gdb` stubs currently: https://github.com/pwndbg/pwndbg/blob/fef5077eef8092cf16828457792f5d402ad792b6/dev-requirements.txt#L10

**Comment by keithasaurus** [comment]
https://github.com/keithasaurus/koda-validate uses a lot of newer parts of the type system and almost all compiler flags:

```python
    Project(
        location="https://github.com/keithasaurus/koda-validate",
        mypy_cmd="{mypy} koda_validate --check-untyped-defs --disallow-any-generics "
                 "--disallow-incomplete-defs --disallow-subclassing-any "
                 "--disallow-untyped-calls --disallow-untyped-decorators "
                 "--disallow-untyped-defs --no-implicit-optional --no-implicit-reexport "
                 "--strict-optional --strict-equality --warn-no-return "
                 "--warn-redundant-casts --warn-return-any --warn-unused-configs "
                 "--warn-unused-ignores --warn-unreachable",
        pip_cmd="{pip} install koda"
    )
```
This worked for me locally. I'd push a PR, but don't have permissions.

**Comment by hauntsaninja** [comment]
https://github.com/Big-Theta/funktools uses new generic syntax

**Comment by hauntsaninja** [comment]
https://github.com/static-frame/static-frame uncovered a crash in mypy https://github.com/python/mypy/issues/17691

**Comment by hauntsaninja** [comment]
https://github.com/viur-framework/viur-toolkit

**Comment by hamdanal** [comment]
I have this project that is a heavy user of numpy, pandas and django: https://github.com/hamdanal/python-stubs. Might be a good candidate to increase numpy coverage. For example, it could have surfaced https://github.com/numpy/numpy/issues/28071
